sbi/inference/base.py

```diff
@@ -152,6 +153,42 @@ def __init__(
             median_observation_distances=[], epochs=[], best_validation_log_probs=[],
         )
 
+    def _append_training_data(
+        self,
+        theta: Tensor,
+        x: Tensor,
+        external_data: Tuple[Tensor, Tensor],
```

Comment:
I would remove `external_data` as keyword argument for this method and handle the concatenation elsewhere

Reply:
I tend to agree here.